// vsts-task-lib/task.js
/// <reference path="../definitions/node.d.ts" />
/// <reference path="../definitions/Q.d.ts" />
/// <reference path="../definitions/shelljs.d.ts" />
/// <reference path="../definitions/minimatch.d.ts" />
/// <reference path="../definitions/glob.d.ts" />
"use strict";
var shell = require('shelljs');
var fs = require('fs');
var path = require('path');
var os = require('os');
var minimatch = require('minimatch');
var globm = require('glob');
var util = require('util');
var tcm = require('./taskcommand');
var trm = require('./toolrunner');
(function (TaskResult) {
    TaskResult[TaskResult["Succeeded"] = 0] = "Succeeded";
    TaskResult[TaskResult["Failed"] = 1] = "Failed";
})(exports.TaskResult || (exports.TaskResult = {}));
var TaskResult = exports.TaskResult;
//-----------------------------------------------------
// String convenience
//-----------------------------------------------------
function startsWith(str, start) {
    return str.slice(0, start.length) == start;
}
function endsWith(str, end) {
    return str.slice(-str.length) == end;
}
//-----------------------------------------------------
// General Helpers
//-----------------------------------------------------
exports._outStream = process.stdout;
exports._errStream = process.stderr;
function _writeError(str) {
    exports._errStream.write(str + os.EOL);
}
exports._writeError = _writeError;
function _writeLine(str) {
    exports._outStream.write(str + os.EOL);
}
exports._writeLine = _writeLine;
function setStdStream(stdStream) {
    exports._outStream = stdStream;
}
exports.setStdStream = setStdStream;
function setErrStream(errStream) {
    exports._errStream = errStream;
}
exports.setErrStream = setErrStream;
//-----------------------------------------------------
// Results and Exiting
//-----------------------------------------------------
/**
 * Sets the result of the task.
 * If the result is Failed (1), then execution will halt.
 *
 * @param result    TaskResult enum of Success or Failed.  If the result is Failed (1), then execution will halt.
 * @param messages  A message which will be logged and added as an issue if an failed
 * @returns         void
 */
function setResult(result, message) {
    debug('task result: ' + TaskResult[result]);
    command('task.complete', { 'result': TaskResult[result] }, message);
    if (result == TaskResult.Failed) {
        _writeError(message);
    }
    if (result == TaskResult.Failed && !process.env['TASKLIB_INPROC_UNITS']) {
        process.exit(0);
    }
}
exports.setResult = setResult;
// deprecated
function handlerError(errMsg, continueOnError) {
    if (continueOnError) {
        error(errMsg);
    }
    else {
        setResult(TaskResult.Failed, errMsg);
    }
}
exports.handlerError = handlerError;
//
// Catching all exceptions
//
process.on('uncaughtException', function (err) {
    setResult(TaskResult.Failed, loc('LIB_UnhandledEx', err.message));
});
// deprecating
function exitOnCodeIf(code, condition) {
    if (condition) {
        setResult(TaskResult.Failed, loc('LIB_FailOnCode', code));
    }
}
exports.exitOnCodeIf = exitOnCodeIf;
//
// back compat: should use setResult
//
function exit(code) {
    setResult(code, loc('LIB_ReturnCode', code));
}
exports.exit = exit;
//-----------------------------------------------------
// Loc Helpers
//-----------------------------------------------------
var locStringCache = {};
var resourceFile;
var libResourceFileLoaded = false;
var resourceCulture = 'en-US';
function loadResJson(resjsonFile) {
    var resJson = null;
    if (exist(resjsonFile)) {
        var resjsonContent = fs.readFileSync(resjsonFile, 'utf8').toString();
        // remove BOM
        if (resjsonContent.indexOf('\uFEFF') == 0) {
            resjsonContent = resjsonContent.slice(1);
        }
        try {
            resJson = JSON.parse(resjsonContent);
        }
        catch (err) {
            debug('unable to parse resjson with err: ' + err.message);
            resJson = null;
        }
    }
    else {
        debug('.resjson file not found: ' + resjsonFile);
    }
    return resJson;
}
function loadLocStrings(resourceFile, culture) {
    var locStrings = {};
    if (exist(resourceFile)) {
        debug('load strings from: ' + resourceFile);
        var resourceJson = require(resourceFile);
        if (resourceJson && resourceJson.hasOwnProperty('messages')) {
            var locResourceJson = null;
            // load up resource resjson for different culture
            var localizedResourceFile = path.join(path.dirname(resourceFile), 'strings', 'resources.resjson', culture, 'resources.resjson');
            if (exist(localizedResourceFile)) {
                debug('load loc strings from: ' + localizedResourceFile);
                locResourceJson = loadResJson(localizedResourceFile);
            }
            for (var key in resourceJson.messages) {
                if (locResourceJson && locResourceJson.hasOwnProperty('loc.messages.' + key)) {
                    locStrings[key] = locResourceJson['loc.messages.' + key];
                }
                else {
                    locStrings[key] = resourceJson.messages[key];
                }
            }
        }
    }
    else {
        warning(loc('LIB_ResourceFileNotExist', resourceFile));
    }
    return locStrings;
}
/**
 * Sets the location of the resources json.  This is typically the task.json file.
 * Call once at the beginning of the script before any calls to loc.
 *
 * @param     path      Full path to the json.
 * @returns   void
 */
function setResourcePath(path) {
    if (process.env['TASKLIB_INPROC_UNITS']) {
        resourceFile = null;
        libResourceFileLoaded = false;
        locStringCache = {};
        resourceCulture = 'en-US';
    }
    if (!resourceFile) {
        checkPath(path, 'resource file path');
        resourceFile = path;
        debug('set resource file to: ' + resourceFile);
        resourceCulture = getVariable('system.culture') || resourceCulture;
        var locStrs = loadLocStrings(resourceFile, resourceCulture);
        for (var key in locStrs) {
            //cache loc string
            locStringCache[key] = locStrs[key];
        }
    }
    else {
        warning(loc('LIB_ResourceFileAlreadySet', resourceFile));
    }
}
exports.setResourcePath = setResourcePath;
/**
 * Gets the localized string from the json resource file.  Optionally formats with additional params.
 *
 * @param     key      key of the resources string in the resource file
 * @param     param    additional params for formatting the string
 * @returns   string
 */
function loc(key) {
    var param = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        param[_i - 1] = arguments[_i];
    }
    if (!libResourceFileLoaded) {
        // merge loc strings from vsts-task-lib.
        var libResourceFile = path.join(__dirname, 'lib.json');
        var libLocStrs = loadLocStrings(libResourceFile, resourceCulture);
        for (var libKey in libLocStrs) {
            //cache vsts-task-lib loc string
            locStringCache[libKey] = libLocStrs[libKey];
        }
        libResourceFileLoaded = true;
    }
    var locString;
    ;
    if (locStringCache.hasOwnProperty(key)) {
        locString = locStringCache[key];
    }
    else {
        if (!resourceFile) {
            warning(loc('LIB_ResourceFileNotSet', key));
        }
        else {
            warning(loc('LIB_LocStringNotFound', key));
        }
        locString = key;
    }
    if (param.length > 0) {
        return util.format.apply(this, [locString].concat(param));
    }
    else {
        return locString;
    }
}
exports.loc = loc;
//-----------------------------------------------------
// Input Helpers
//-----------------------------------------------------
/**
 * Gets a variables value which is defined on the build definition or set at runtime.
 *
 * @param     name     name of the variable to get
 * @returns   string
 */
function getVariable(name) {
    var varval = process.env[name.replace(/\./g, '_').toUpperCase()];
    debug(name + '=' + varval);
    return varval;
}
exports.getVariable = getVariable;
/**
 * Sets a variables which will be available to subsequent tasks as well.
 *
 * @param     name     name of the variable to set
 * @param     val     value to set
 * @returns   void
 */
function setVariable(name, val) {
    if (!name) {
        setResult(TaskResult.Failed, loc('LIB_ParameterIsRequired', 'name'));
    }
    var varValue = val || '';
    process.env[name.replace(/\./g, '_').toUpperCase()] = varValue;
    debug('set ' + name + '=' + varValue);
    command('task.setvariable', { 'variable': name || '' }, varValue);
}
exports.setVariable = setVariable;
/**
 * Gets the value of an input.  The value is also trimmed.
 * If required is true and the value is not set, the task will fail with an error.  Execution halts.
 *
 * @param     name     name of the input to get
 * @param     required whether input is required.  optional, defaults to false
 * @returns   string
 */
function getInput(name, required) {
    var inval = process.env['INPUT_' + name.replace(' ', '_').toUpperCase()];
    if (inval) {
        inval = inval.trim();
    }
    if (required && !inval) {
        setResult(TaskResult.Failed, loc('LIB_InputRequired', name));
    }
    debug(name + '=' + inval);
    return inval;
}
exports.getInput = getInput;
/**
 * Gets the value of an input and converts to a bool.  Convenience.
 * If required is true and the value is not set, the task will fail with an error.  Execution halts.
 *
 * @param     name     name of the bool input to get
 * @param     required whether input is required.  optional, defaults to false
 * @returns   string
 */
function getBoolInput(name, required) {
    return getInput(name, required) == "true";
}
exports.getBoolInput = getBoolInput;
// deprecated - use  setVariable
function setEnvVar(name, val) {
    if (val) {
        process.env[name] = val;
    }
}
exports.setEnvVar = setEnvVar;
/**
 * Gets the value of an input and splits the values by a delimiter (space, comma, etc...)
 * Useful for splitting an input with simple list of items like targets
 * IMPORTANT: Do not use for splitting additional args!  Instead use arg() - it will split and handle
 * If required is true and the value is not set, the task will fail with an error.  Execution halts.
 *
 * @param     name     name of the input to get
 * @param     delim     delimiter to split on
 * @param     required whether input is required.  optional, defaults to false
 * @returns   string[]
 */
function getDelimitedInput(name, delim, required) {
    var inval = getInput(name, required);
    if (!inval) {
        return [];
    }
    return inval.split(delim);
}
exports.getDelimitedInput = getDelimitedInput;
/**
 * Checks whether a path inputs value was supplied by the user
 * File paths are relative with a picker, so an empty path is the root of the repo.
 * Useful if you need to condition work (like append an arg) if a value was supplied
 *
 * @param     name      name of the path input to check
 * @returns   boolean
 */
function filePathSupplied(name) {
    // normalize paths
    var pathValue = path.resolve(this.getPathInput(name) || '');
    var repoRoot = path.resolve(this.getVariable('build.sourcesDirectory') || '');
    var supplied = pathValue !== repoRoot;
    debug(name + 'path supplied :' + supplied);
    return supplied;
}
exports.filePathSupplied = filePathSupplied;
/**
 * Gets the value of a path input
 * It will be quoted for you if it isn't already and contains spaces
 * If required is true and the value is not set, the task will fail with an error.  Execution halts.
 * If check is true and the path does not exist, the task will fail with an error.  Execution halts.
 *
 * @param     name      name of the input to get
 * @param     required  whether input is required.  optional, defaults to false
 * @param     check     whether path is checked.  optional, defaults to false
 * @returns   string
 */
function getPathInput(name, required, check) {
    var inval = getInput(name, required);
    if (inval) {
        if (check) {
            checkPath(inval, name);
        }
    }
    return inval;
}
exports.getPathInput = getPathInput;
//-----------------------------------------------------
// Endpoint Helpers
//-----------------------------------------------------
/**
 * Gets the url for a service endpoint
 * If the url was not set and is not optional, the task will fail with an error message. Execution will halt.
 *
 * @param     id        name of the service endpoint
 * @param     optional  whether the url is optional
 * @returns   string
 */
function getEndpointUrl(id, optional) {
    var urlval = process.env['ENDPOINT_URL_' + id];
    if (!optional && !urlval) {
        setResult(TaskResult.Failed, loc('LIB_EndpointNotExist', id));
    }
    debug(id + '=' + urlval);
    return urlval;
}
exports.getEndpointUrl = getEndpointUrl;
/**
 * Gets the authorization details for a service endpoint
 * If the authorization was not set and is not optional, the task will fail with an error message. Execution will halt.
 *
 * @param     id        name of the service endpoint
 * @param     optional  whether the url is optional
 * @returns   string
 */
function getEndpointAuthorization(id, optional) {
    var aval = process.env['ENDPOINT_AUTH_' + id];
    if (!optional && !aval) {
        setResult(TaskResult.Failed, loc('LIB_EndpointNotExist', id));
    }
    debug(id + '=' + aval);
    var auth;
    try {
        auth = JSON.parse(aval);
    }
    catch (err) {
        setResult(TaskResult.Failed, loc('LIB_InvalidEndpointAuth', aval)); // exit
    }
    return auth;
}
exports.getEndpointAuthorization = getEndpointAuthorization;
//-----------------------------------------------------
// Cmd Helpers
//-----------------------------------------------------
function command(command, properties, message) {
    var taskCmd = new tcm.TaskCommand(command, properties, message);
    _writeLine(taskCmd.toString());
}
exports.command = command;
function warning(message) {
    command('task.issue', { 'type': 'warning' }, message);
}
exports.warning = warning;
function error(message) {
    command('task.issue', { 'type': 'error' }, message);
}
exports.error = error;
function debug(message) {
    command('task.debug', null, message);
}
exports.debug = debug;
/**
 * Get's stat on a path.
 * Useful for checking whether a file or directory.  Also getting created, modified and accessed time.
 * see [fs.stat](https://nodejs.org/api/fs.html#fs_class_fs_stats)
 *
 * @param     path      path to check
 * @returns   fsStat
 */
function stats(path) {
    return fs.statSync(path);
}
exports.stats = stats;
/**
 * Returns whether a path exists.
 *
 * @param     path      path to check
 * @returns   boolean
 */
function exist(path) {
    return path && fs.existsSync(path);
}
exports.exist = exist;
/**
 * Checks whether a path exists.
 * If the path does not exist, the task will fail with an error message. Execution will halt.
 *
 * @param     p         path to check
 * @param     name      name only used in error message to identify the path
 * @returns   void
 */
function checkPath(p, name) {
    debug('check path : ' + p);
    if (!exist(p)) {
        setResult(TaskResult.Failed, loc('LIB_PathNotFound', name, p)); // exit
    }
}
exports.checkPath = checkPath;
/**
 * Change working directory.
 *
 * @param     path      new working directory path
 * @returns   void
 */
function cd(path) {
    if (path) {
        shell.cd(path);
    }
}
exports.cd = cd;
/**
 * Change working directory and push it on the stack
 *
 * @param     path      new working directory path
 * @returns   void
 */
function pushd(path) {
    shell.pushd(path);
}
exports.pushd = pushd;
/**
 * Change working directory back to previously pushed directory
 *
 * @returns   void
 */
function popd() {
    shell.popd();
}
exports.popd = popd;
/**
 * Make a directory.  Creates the full path with folders in between
 * Returns whether it was successful or not
 *
 * @param     p       path to create
 * @returns   boolean
 */
function mkdirP(p) {
    var success = true;
    try {
        if (!p) {
            throw new Error(loc('LIB_ParameterIsRequired', 'p'));
        }
        // certain chars like \0 will cause shelljs and fs
        // to blow up without exception or error
        if (p.indexOf('\0') >= 0) {
            throw new Error(loc('LIB_PathHasNullByte'));
        }
        if (!shell.test('-d', p)) {
            debug('creating path: ' + p);
            shell.mkdir('-p', p);
            var errMsg = shell.error();
            if (errMsg) {
                handlerError(errMsg, false);
                success = false;
            }
        }
        else {
            debug('path exists: ' + p);
        }
    }
    catch (err) {
        success = false;
        handlerError(loc('LIB_OperationFailed', 'mkdirP', err.message), false);
    }
    return success;
}
exports.mkdirP = mkdirP;
/**
 * Returns path of a tool had the tool actually been invoked.  Resolves via paths.
 * If you check and the tool does not exist, the task will fail with an error message and halt execution.
 *
 * @param     tool       name of the tool
 * @param     check      whether to check if tool exists
 * @returns   string
 */
function which(tool, check) {
    try {
        // we can't use shelljs.which() on windows due to https://github.com/shelljs/shelljs/issues/238
        // shelljs.which() does not prefer file with executable extensions (.exe, .bat, .cmd).
        // we already made a PR for Shelljs, but they haven't merge it yet. https://github.com/shelljs/shelljs/pull/239
        if (os.type().match(/^Win/)) {
            var pathEnv = process.env.path || process.env.Path || process.env.PATH;
            var pathArray = pathEnv.split(';');
            var toolPath = null;
            // No relative/absolute paths provided?
            if (tool.search(/[\/\\]/) === -1) {
                // Search for command in PATH
                pathArray.forEach(function (dir) {
                    if (toolPath)
                        return; // already found it
                    var attempt = path.resolve(dir + '/' + tool);
                    var baseAttempt = attempt;
                    attempt = baseAttempt + '.exe';
                    if (exist(attempt) && stats(attempt).isFile) {
                        toolPath = attempt;
                        return;
                    }
                    attempt = baseAttempt + '.cmd';
                    if (exist(attempt) && stats(attempt).isFile) {
                        toolPath = attempt;
                        return;
                    }
                    attempt = baseAttempt + '.bat';
                    if (exist(attempt) && stats(attempt).isFile) {
                        toolPath = attempt;
                        return;
                    }
                });
            }
            // Command not found in Path, but the input itself is point to a file.
            if (!toolPath && exist(tool) && stats(tool).isFile) {
                toolPath = path.resolve(tool);
            }
        }
        else {
            var toolPath = shell.which(tool);
        }
        if (check) {
            checkPath(toolPath, tool);
        }
        debug(tool + '=' + toolPath);
        return toolPath;
    }
    catch (err) {
        handlerError(loc('LIB_OperationFailed', 'which', err.message), false);
    }
}
exports.which = which;
/**
 * Returns path of a tool had the tool actually been invoked.  Resolves via paths.
 * If you check and the tool does not exist, the task will fail with an error message and halt execution.
 * Returns whether the copy was successful
 *
 * @param     options    string -r, -f or -rf for recursive and force
 * @param     source     source path
 * @param     dest       destination path
 * @param     continueOnError optional. whether to continue on error
 * @returns   boolean
 */
function cp(options, source, dest, continueOnError) {
    var success = true;
    try {
        shell.cp(options, source, dest);
        var errMsg = shell.error();
        if (errMsg) {
            handlerError(errMsg, continueOnError);
            success = false;
        }
    }
    catch (err) {
        success = false;
        handlerError(loc('LIB_OperationFailed', 'cp', err.message), false);
    }
    return success;
}
exports.cp = cp;
/**
 * Moves a path.
 * Returns whether the copy was successful
 *
 * @param     source     source path
 * @param     dest       destination path
 * @param     force      whether to force and overwrite
 * @param     continueOnError optional. whether to continue on error
 * @returns   boolean
 */
function mv(source, dest, force, continueOnError) {
    var success = true;
    try {
        if (force) {
            shell.mv('-f', source, dest);
        }
        else {
            shell.mv(source, dest);
        }
        var errMsg = shell.error();
        if (errMsg) {
            handlerError(errMsg, continueOnError);
            success = false;
        }
    }
    catch (err) {
        success = false;
        handlerError(loc('LIB_OperationFailed', 'mv', err.message), false);
    }
    return success;
}
exports.mv = mv;
/**
 * Find all files under a give path
 * Returns an array of full paths
 *
 * @param     findPath     path to find files under
 * @returns   string[]
 */
function find(findPath) {
    try {
        if (!shell.test('-e', findPath)) {
            return [];
        }
        var matches = shell.find(findPath);
        debug('find ' + findPath);
        debug(matches.length + ' matches.');
        return matches;
    }
    catch (err) {
        handlerError(loc('LIB_OperationFailed', 'find', err.message), false);
    }
}
exports.find = find;
/**
 * Remove a path recursively with force
 * Returns whether it succeeds
 *
 * @param     path     path to remove
 * @param     continueOnError optional. whether to continue on error
 * @returns   string[]
 */
function rmRF(path, continueOnError) {
    var success = true;
    try {
        debug('rm -rf ' + path);
        shell.rm('-rf', path);
        var errMsg = shell.error();
        // if you try to delete a file that doesn't exist, desired result is achieved
        // other errors are valid
        if (errMsg && !(errMsg.indexOf('ENOENT') === 0)) {
            handlerError(errMsg, continueOnError);
            success = false;
        }
    }
    catch (err) {
        success = false;
        handlerError(loc('LIB_OperationFailed', 'rmRF', err.message), false);
    }
    return success;
}
exports.rmRF = rmRF;
function glob(pattern) {
    debug('glob ' + pattern);
    var matches = globm.sync(pattern);
    debug('found ' + matches.length + ' matches');
    if (matches.length > 0) {
        var m = Math.min(matches.length, 10);
        debug('matches:');
        if (m == 10) {
            debug('listing first 10 matches as samples');
        }
        for (var i = 0; i < m; i++) {
            debug(matches[i]);
        }
    }
    return matches;
}
exports.glob = glob;
function globFirst(pattern) {
    debug('globFirst ' + pattern);
    var matches = glob(pattern);
    if (matches.length > 1) {
        warning(loc('LIB_UseFirstGlobMatch'));
    }
    debug('found ' + matches.length + ' matches');
    return matches[0];
}
exports.globFirst = globFirst;
//-----------------------------------------------------
// Exec convenience wrapper
//-----------------------------------------------------
var _argStringToArray = function (argString) {
    var args = argString.match(/([^" ]*("[^"]*")[^" ]*)|[^" ]+/g);
    for (var i = 0; i < args.length; i++) {
        args[i] = args[i].replace(/"/g, "");
    }
    return args;
};
/**
 * Exec a tool.  Convenience wrapper over ToolRunner to exec with args in one call.
 * Output will be streamed to the live console.
 * Returns promise with return code
 *
 * @param     tool     path to tool to exec
 * @param     args     an arg string or array of args
 * @param     options  optional exec options.  See IExecOptions
 * @returns   number
 */
function exec(tool, args, options) {
    var toolPath = which(tool, true);
    var tr = createToolRunner(toolPath);
    if (args) {
        tr.arg(args);
    }
    return tr.exec(options);
}
exports.exec = exec;
/**
 * Exec a tool synchronously.  Convenience wrapper over ToolRunner to execSync with args in one call.
 * Output will be *not* be streamed to the live console.  It will be returned after execution is complete.
 * Appropriate for short running tools
 * Returns IExecResult with output and return code
 *
 * @param     tool     path to tool to exec
 * @param     args     an arg string or array of args
 * @param     options  optionalexec options.  See IExecOptions
 * @returns   IExecResult
 */
function execSync(tool, args, options) {
    var toolPath = which(tool, true);
    var tr = createToolRunner(toolPath);
    if (args) {
        tr.arg(args);
    }
    return tr.execSync(options);
}
exports.execSync = execSync;
/**
 * Convenience factory to create a ToolRunner.
 *
 * @param     tool     path to tool to exec
 * @returns   ToolRunner
 */
function createToolRunner(tool) {
    var tr = new trm.ToolRunner(tool);
    tr.on('debug', function (message) {
        debug(message);
    });
    return tr;
}
exports.createToolRunner = createToolRunner;
//-----------------------------------------------------
// Matching helpers
//-----------------------------------------------------
function match(list, pattern, options) {
    return minimatch.match(list, pattern, options);
}
exports.match = match;
function filter(pattern, options) {
    return minimatch.filter(pattern, options);
}
exports.filter = filter;
//-----------------------------------------------------
// Test Publisher
//-----------------------------------------------------
var TestPublisher = (function () {
    function TestPublisher(testRunner) {
        this.testRunner = testRunner;
    }
    TestPublisher.prototype.publish = function (resultFiles, mergeResults, platform, config, runTitle, publishRunAttachments) {
        if (mergeResults == 'true') {
            _writeLine(loc('LIB_MergeTestResultNotSupported'));
        }
        var properties = {};
        properties['type'] = this.testRunner;
        if (platform) {
            properties['platform'] = platform;
        }
        if (config) {
            properties['config'] = config;
        }
        if (runTitle) {
            properties['runTitle'] = runTitle;
        }
        if (publishRunAttachments) {
            properties['publishRunAttachments'] = publishRunAttachments;
        }
        for (var i = 0; i < resultFiles.length; i++) {
            properties['fileNumber'] = i.toString();
            command('results.publish', properties, resultFiles[i]);
        }
    };
    return TestPublisher;
}());
exports.TestPublisher = TestPublisher;
//-----------------------------------------------------
// Tools
//-----------------------------------------------------
exports.TaskCommand = tcm.TaskCommand;
exports.commandFromString = tcm.commandFromString;
exports.ToolRunner = trm.ToolRunner;
trm.debug = debug;
