// vso-node-api/interfaces/common/OperationsInterfaces.js
/*
* ---------------------------------------------------------
* Copyright(C) Microsoft Corporation. All rights reserved.
* ---------------------------------------------------------
*
* ---------------------------------------------------------
* Generated file, DO NOT EDIT
* ---------------------------------------------------------
*/
"use strict";
(function (OperationStatus) {
    OperationStatus[OperationStatus["NotSet"] = 0] = "NotSet";
    OperationStatus[OperationStatus["Queued"] = 1] = "Queued";
    OperationStatus[OperationStatus["InProgress"] = 2] = "InProgress";
    OperationStatus[OperationStatus["Cancelled"] = 3] = "Cancelled";
    OperationStatus[OperationStatus["Succeeded"] = 4] = "Succeeded";
    OperationStatus[OperationStatus["Failed"] = 5] = "Failed";
})(exports.OperationStatus || (exports.OperationStatus = {}));
var OperationStatus = exports.OperationStatus;
exports.TypeInfo = {
    OperationReference: {
        fields: null
    },
    OperationStatus: {
        enumValues: {
            "notSet": 0,
            "queued": 1,
            "inProgress": 2,
            "cancelled": 3,
            "succeeded": 4,
            "failed": 5
        }
    }
};
exports.TypeInfo.OperationReference.fields = {
    status: {
        enumType: exports.TypeInfo.OperationStatus
    }
};
