// vso-node-api/interfaces/common/SystemDataInterfaces.js
//----------------------------------------------------------
// Copyright (C) Microsoft Corporation. All rights reserved.
//----------------------------------------------------------
(function (SqlDbType) {
    SqlDbType[SqlDbType["BigInt"] = 0] = "BigInt";
    SqlDbType[SqlDbType["Binary"] = 1] = "Binary";
    SqlDbType[SqlDbType["Bit"] = 2] = "Bit";
    SqlDbType[SqlDbType["Char"] = 3] = "Char";
    SqlDbType[SqlDbType["DateTime"] = 4] = "DateTime";
    SqlDbType[SqlDbType["Decimal"] = 5] = "Decimal";
    SqlDbType[SqlDbType["Float"] = 6] = "Float";
    SqlDbType[SqlDbType["Image"] = 7] = "Image";
    SqlDbType[SqlDbType["Int"] = 8] = "Int";
    SqlDbType[SqlDbType["Money"] = 9] = "Money";
    SqlDbType[SqlDbType["NChar"] = 10] = "NChar";
    SqlDbType[SqlDbType["NText"] = 11] = "NText";
    SqlDbType[SqlDbType["NVarChar"] = 12] = "NVarChar";
    SqlDbType[SqlDbType["Real"] = 13] = "Real";
    SqlDbType[SqlDbType["UniqueIdentifier"] = 14] = "UniqueIdentifier";
    SqlDbType[SqlDbType["SmallDateTime"] = 15] = "SmallDateTime";
    SqlDbType[SqlDbType["SmallInt"] = 16] = "SmallInt";
    SqlDbType[SqlDbType["SmallMoney"] = 17] = "SmallMoney";
    SqlDbType[SqlDbType["Text"] = 18] = "Text";
    SqlDbType[SqlDbType["Timestamp"] = 19] = "Timestamp";
    SqlDbType[SqlDbType["TinyInt"] = 20] = "TinyInt";
    SqlDbType[SqlDbType["VarBinary"] = 21] = "VarBinary";
    SqlDbType[SqlDbType["VarChar"] = 22] = "VarChar";
    SqlDbType[SqlDbType["Variant"] = 23] = "Variant";
    SqlDbType[SqlDbType["Xml"] = 25] = "Xml";
    SqlDbType[SqlDbType["Udt"] = 29] = "Udt";
    SqlDbType[SqlDbType["Structured"] = 30] = "Structured";
    SqlDbType[SqlDbType["Date"] = 31] = "Date";
    SqlDbType[SqlDbType["Time"] = 32] = "Time";
    SqlDbType[SqlDbType["DateTime2"] = 33] = "DateTime2";
    SqlDbType[SqlDbType["DateTimeOffset"] = 34] = "DateTimeOffset";
})(exports.SqlDbType || (exports.SqlDbType = {}));
var SqlDbType = exports.SqlDbType;
exports.TypeInfo = {
    SqlDbType: {
        enumValues: {
            "BigInt": 0,
            "Binary": 1,
            "Bit": 2,
            "Char": 3,
            "DateTime": 4,
            "Decimal": 5,
            "Float": 6,
            "Image": 7,
            "Int": 8,
            "Money": 9,
            "NChar": 10,
            "NText": 11,
            "NVarChar": 12,
            "Real": 13,
            "UniqueIdentifier": 14,
            "SmallDateTime": 15,
            "SmallInt": 16,
            "SmallMoney": 17,
            "Text": 18,
            "Timestamp": 19,
            "TinyInt": 20,
            "VarBinary": 21,
            "VarChar": 22,
            "Variant": 23,
            "Xml": 25,
            "Udt": 29,
            "Structured": 30,
            "Date": 31,
            "Time": 32,
            "DateTime2": 33,
            "DateTimeOffset": 34
        }
    }
};
