// vso-node-api/interfaces/common/VSSInterfaces.js
/*
* ---------------------------------------------------------
* Copyright(C) Microsoft Corporation. All rights reserved.
* ---------------------------------------------------------
*
* ---------------------------------------------------------
* Generated file, DO NOT EDIT
* ---------------------------------------------------------
*/
"use strict";
(function (Operation) {
    Operation[Operation["Add"] = 0] = "Add";
    Operation[Operation["Remove"] = 1] = "Remove";
    Operation[Operation["Replace"] = 2] = "Replace";
    Operation[Operation["Move"] = 3] = "Move";
    Operation[Operation["Copy"] = 4] = "Copy";
    Operation[Operation["Test"] = 5] = "Test";
})(exports.Operation || (exports.Operation = {}));
var Operation = exports.Operation;
exports.TypeInfo = {
    IdentityRef: {
        fields: null
    },
    JsonPatchDocument: {
        fields: null
    },
    JsonPatchOperation: {
        fields: null
    },
    Operation: {
        enumValues: {
            "add": 0,
            "remove": 1,
            "replace": 2,
            "move": 3,
            "copy": 4,
            "test": 5
        }
    },
    OperationReference: {
        fields: null
    },
    ResourceRef: {
        fields: null
    },
    VssJsonCollectionWrapper: {
        fields: null
    },
    VssJsonCollectionWrapperV: {
        fields: null
    },
    VssJsonCollectionWrapperBase: {
        fields: null
    }
};
exports.TypeInfo.IdentityRef.fields = {};
exports.TypeInfo.JsonPatchDocument.fields = {};
exports.TypeInfo.JsonPatchOperation.fields = {
    op: {
        enumType: exports.TypeInfo.Operation
    }
};
exports.TypeInfo.OperationReference.fields = {};
exports.TypeInfo.ResourceRef.fields = {};
exports.TypeInfo.VssJsonCollectionWrapper.fields = {};
exports.TypeInfo.VssJsonCollectionWrapperV.fields = {};
exports.TypeInfo.VssJsonCollectionWrapperBase.fields = {};
