// vso-node-api/interfaces/FileContainerInterfaces.js
/*
* ---------------------------------------------------------
* Copyright(C) Microsoft Corporation. All rights reserved.
* ---------------------------------------------------------
*
* ---------------------------------------------------------
* Generated file, DO NOT EDIT
* ---------------------------------------------------------
*/
"use strict";
(function (ContainerItemStatus) {
    ContainerItemStatus[ContainerItemStatus["Created"] = 1] = "Created";
    ContainerItemStatus[ContainerItemStatus["PendingUpload"] = 2] = "PendingUpload";
})(exports.ContainerItemStatus || (exports.ContainerItemStatus = {}));
var ContainerItemStatus = exports.ContainerItemStatus;
(function (ContainerItemType) {
    ContainerItemType[ContainerItemType["Any"] = 0] = "Any";
    ContainerItemType[ContainerItemType["Folder"] = 1] = "Folder";
    ContainerItemType[ContainerItemType["File"] = 2] = "File";
})(exports.ContainerItemType || (exports.ContainerItemType = {}));
var ContainerItemType = exports.ContainerItemType;
(function (ContainerOptions) {
    ContainerOptions[ContainerOptions["None"] = 0] = "None";
})(exports.ContainerOptions || (exports.ContainerOptions = {}));
var ContainerOptions = exports.ContainerOptions;
exports.TypeInfo = {
    ContainerItemStatus: {
        enumValues: {
            "created": 1,
            "pendingUpload": 2
        }
    },
    ContainerItemType: {
        enumValues: {
            "any": 0,
            "folder": 1,
            "file": 2
        }
    },
    ContainerOptions: {
        enumValues: {
            "none": 0
        }
    },
    FileContainer: {
        fields: null
    },
    FileContainerItem: {
        fields: null
    }
};
exports.TypeInfo.FileContainer.fields = {
    dateCreated: {
        isDate: true
    },
    options: {
        enumType: exports.TypeInfo.ContainerOptions
    }
};
exports.TypeInfo.FileContainerItem.fields = {
    dateCreated: {
        isDate: true
    },
    dateLastModified: {
        isDate: true
    },
    itemType: {
        enumType: exports.TypeInfo.ContainerItemType
    },
    status: {
        enumType: exports.TypeInfo.ContainerItemStatus
    }
};
